orion/pkg/domains/hash.go

```diff
@@ -0,0 +1,23 @@
+package domains
+
+import (
+	"golang.org/x/crypto/bcrypt"
+)
+
+type Hash struct {
+	HashBytes []byte
+}
+
+func (hash *Hash) GetHash(str string) error {
```

Comment:
I think having Compare be part of the Hash struct makes sense. But for creating, I don't think it should be a struct method. It can be a standalone utility function that returns a hash, and/or an error. See my comment below in the test file.